### PR TITLE
enh(drawio): enhance editor links management

### PR DESCRIPTION
**Pull request by Yassir-BenBOUBKER**
delete the orange circle from sidebar shape
image

### CODE REVIEW

**Comment by Yassir-BenBOUBKER** [comment]
https://user-images.githubusercontent.com/97686091/172863726-9495bcb3-348e-4df9-b75b-b2e62b87d04a.mp4

File: src/main/webapp/js/diagramly/EditorUi.js [hunk 1]
```diff
@@ -12924,6 +12924,10 @@
 
 						graph.getModel().setValue(cell, cellValue);
 
+						if(graph.model.isEdge(cell) && cell.source && cell.target) {
```
Comment (Thebarda) [inline src/main/webapp/js/diagramly/EditorUi.js]:
```suggestion
						if (graph.model.isEdge(cell) && cell.source && cell.target) {
```

File: src/main/webapp/js/grapheditor/Graph.js [hunk 2]
```diff
@@ -2625,6 +2645,32 @@ Graph.prototype.init = function(container)
 		return cells;
 	};
 
+	Graph.prototype.isDataAddedToEdge = function(edge){
+		console.log('I am in isDataAddedToEdge');
```
Comment (bdauria) [inline src/main/webapp/js/grapheditor/Graph.js]: Remove this

Comment (jdelpierre) [inline src/main/webapp/js/grapheditor/Graph.js]: remove debug console.log()

File: src/main/webapp/js/grapheditor/Actions.js [hunk 3]
```diff
@@ -414,6 +414,26 @@ Actions.prototype.init = function()
...
+
+					if (edges != null && edges.length > 0)
+					{
+						graph.setCellStyles(mxConstants.STYLE_DASHED, '1', edges);
+					}
+				}
+			})
+
+		}
+
```
Comment (Thebarda) [inline src/main/webapp/js/grapheditor/Actions.js]:
```suggestion
		if (cells != null && cells.length > 0)
		{
			cells.forEach((cell) => {
				if (!graph.model.isEdge(cell)) {
					const edges = graph.getEdges(cell);
					if (edges !== null && edges.length > 0)
					{
						graph.setCellStyles(mxConstants.STYLE_DASHED, '1', edges);
					}
				}
			})
		}

```

File: src/main/webapp/js/grapheditor/Graph.js (hunk 2 above)
Comment (Thebarda) [inline src/main/webapp/js/grapheditor/Graph.js]:
```suggestion
```

File: src/main/webapp/js/grapheditor/Graph.js [hunk 4]
```diff
@@ -2625,6 +2645,32 @@ Graph.prototype.init = function(container)
...
+		const edgeTypes = ['METRIC','STATUS']
+		const linkType = edge.getAttribute('linkType');
+		let idDateAdded = false;
+
+		if (edge.getAttribute('resourceId') && edgeTypes.includes(linkType)) {
+			idDateAdded = true;
+		}
+
+		return idDateAdded;
```
Comment (Thebarda) [inline src/main/webapp/js/grapheditor/Graph.js]:
```suggestion
		return edge.getAttribute('resourceId') && edgeTypes.includes(linkType)
```

File: src/main/webapp/js/grapheditor/Graph.js [hunk 5]
```diff
@@ -2625,6 +2645,32 @@ Graph.prototype.init = function(container)
...
+		const connectedEdge =  mxConnectCellEvent.apply(this, arguments);
+
+		if((this.isDataAddedToEdge(connectedEdge) || connectedEdge.getAttribute('linkType') == 'SIMPLE') && connectedEdge.source && connectedEdge.target) {
+			this.setCellStyles(mxConstants.STYLE_DASHED, '0', [edge]);
+		}else if(this.getCellStyle(connectedEdge).dashed === 0) {
+			this.setCellStyles(mxConstants.STYLE_DASHED, '1', [edge]);
+		}		
+
+		return connectedEdge;
+	}
```
Comment (Thebarda) [inline src/main/webapp/js/grapheditor/Graph.js]:
```suggestion
		if ((this.isDataAddedToEdge(connectedEdge) || connectedEdge.getAttribute('linkType') === 'SIMPLE') && connectedEdge.source && connectedEdge.target) {
			this.setCellStyles(mxConstants.STYLE_DASHED, '0', [edge]);
			return connectedEdge;
		}
		
		if(this.getCellStyle(connectedEdge).dashed === 0) {
			this.setCellStyles(mxConstants.STYLE_DASHED, '1', [edge]);
		}		

		return connectedEdge;
	}
```

File: src/main/webapp/js/grapheditor/Sidebar.js [hunk 6]
```diff
@@ -3485,6 +3485,33 @@ Sidebar.prototype.createDragSource = function(elt, dropHandler, preview, cells,
...
 {
 	var graph = this.editorUi.editor.graph;
+	const contextualMenuCellType = ['RESOURCE', 'WIDGET'];
+	cells.forEach((cell) => {
+		if(graph.model.isEdge(cell)) {
+			graph.setCellStyles(mxConstants.STYLE_DASHED, '1', [cell]);
+		}
+
+		const typeCell = cell.getAttribute('type');
+		if(!contextualMenuCellType.includes(typeCell)) {
```
Comment (Thebarda) [inline src/main/webapp/js/grapheditor/Sidebar.js]:
```suggestion
		if (graph.model.isEdge(cell)) {
			graph.setCellStyles(mxConstants.STYLE_DASHED, '1', [cell]);
		}

		const typeCell = cell.getAttribute('type');
		if (!contextualMenuCellType.includes(typeCell)) {
```

**Review by bdauria — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**